### PR TITLE
[v2.10] k8s November 2024 Patches 

### PR DESCRIPTION
**Pull request by krunalhinguu**
### Updates

- Added k8s version 1.29.11 for November Patches
- Added k8s version 1.30.7 for November Patches
- Added k8s version 1.31.3 for November Patches

### Related

- https://github.com/rancher/rancher/issues/48083

### CODE REVIEW

File: pkg/rke/templates/calico_v3.29.0.go [hunk 1]
```diff
@@ -0,0 +1,6529 @@
...
+                      type: string
+                    interface:
+                      type: string
+                    matchOperator:
+                      type: string
+                    prefixLength:
+                      properties:
+                        max:
+                          format: int32
+                          maximum: 32
```
Comment (kinarashah) [inline pkg/rke/templates/calico_v3.29.0.go]: Something seems off here. Upstream's maximum is set to a different value, was this not updated earlier or it's not picked from upstream now? https://raw.githubusercontent.com/projectcalico/calico/v3.29.0/manifests/canal.yaml

File: pkg/rke/templates/calico_v3.29.0.go [hunk 2]
```diff
@@ -0,0 +1,6529 @@
...
+                          minimum: 0
+                          type: integer
+                        min:
+                          format: int32
+                          maximum: 32
```
Comment (kinarashah) [inline pkg/rke/templates/calico_v3.29.0.go]: Same as above. https://www.diffchecker.com/1XGIbsAP/ for the full diff.

**Review by kinarashah — CHANGES_REQUESTED** [review]

**Comment by krunalhinguu** [comment]
updated @kinarashah

**Review by kinarashah — CHANGES_REQUESTED** [review]
@krunalhinguu Same changes required for canal too

**Review by kinarashah — APPROVED** [review]

**Pull request merged**